examples/wasm/envoy_filter_http_wasm_example.cc

```diff
@@ -1,14 +1,221 @@
+   if (field.is_null()) {
+     return true;
+   }
+   if (!field["include"].is_null() && !field["exclude"].is_null()) {
+     LOG_ERROR("onConfigure: \"include\" and \"exclude\" should not be used simultaneously");
+     return false;
+   }
+   if (!field["include"].is_null()) {
+     *include = true;
+     auto include_params = field["include"].get<std::vector<std::string>>();
+     params->insert(params->end(), include_params.begin(), include_params.end());
+   }
+   if (!field["exclude"].is_null()) {
+     *include = false;
+     *params = field["exclude"].get<std::vector<std::string>>();
+   }
+   return true;
+}
+
+/*
+ * Methods for parsing query parameters
+ */
+QueryParams parseParameters(std::string data, size_t start,
+                            std::string delim = "&", std::string eq = "=") {
+  QueryParams params;
+
+  while (start < data.size()) {
+    size_t end = data.find(delim, start);
+    if (end == std::string::npos) {
+      end = data.size();
+    }
+    std::string param = data.substr(start, end - start);
+
+    const size_t equal = param.find(eq);
+    if (equal != std::string::npos) {
+      params.emplace(param.substr(start, start + equal),
+                     param.substr(start + equal + 1, end));
+    } else {
+      params.emplace(param.substr(start, end), "");
+    }
+
+    start = end + 1;
+  }
+
+  return params;
+}
+
+QueryParams parsePath(std::string path) {
+  size_t start = path.find('?');
+  if (start == std::string::npos) {
+    QueryParams params;
+    return params;
+  }
+  start++;
+  return parseParameters(path, start);
+}
+
+QueryParams parseBody(std::string body) {
+  return parseParameters(body, 0);
+}
+
+QueryParams parseCookie(std::string cookie) {
+  return parseParameters(cookie, 0, "; ");
+}
+
+std::string toString(QueryParams params) {
+  std::string str;
+  for (auto param : params) {
+    str += param.first + " -> " + param.second;
+  }
+  return str;
+}
+
+/*
+ * Detect SQL injection on given input string
+ */
+bool detectSQLi(std::string input, std::string key, std::string part) {
+  struct libinjection_sqli_state state;
+  char* input_char_str = const_cast<char*>(input.c_str());
+  libinjection_sqli_init(&state, input_char_str, input.length(), FLAG_NONE);
+
+  int issqli = libinjection_is_sqli(&state);
+  if (issqli) {
+    std::string response_body = "SQL injection detected";
+    std::string response_log = "SQLi at " + part + "->" + key + ", fingerprint: "
+        + std::string(state.fingerprint);
+    sendLocalResponse(403, response_log, response_body, {});
+    return true;
+  }
+  LOG_TRACE("detectSQLi: " + part + "->" + key + " passed detection");
+  return false;
+}
+
+/**
+ * Detect SQL injection on given parameter pairs with configuration
+ * Input
+ *  - params: a map of param key value pairs
+ *  - include: a boolean
+ *      if true, given keys are the only keys to detect
+ *      if false, given keys are all but the given keys will be detected
```

Comment:
the comment needs a bit update, not clear what does it mean: ` given keys are all but the given keys will be detected`?

Reply:
That was a typo! I was hoping to say ```if true, given keys are the only keys to detect
if false, all but the given keys will be detected```. Do I need to further explain that?